### PR TITLE
[s2s] configure lr_scheduler from command line

### LINKED ISSUE #7543: Seq2SeqTrainer: missing features
These could all be separate issues, if you want to tackle 1 feel free to make a new issue to link to your PR, or not!

1. Configure lr scheduler from the command line https://github.com/huggingface/transformers/blob/master/examples/lightning_base.py#L119
2. Configure dropout, layerdrop from the command line: https://github.com/huggingface/transformers/blob/master/examples/lightning_base.py#L92
3. Logging to wandb seems much less frequent than with the PL integration, which sends train loss to wandb every step.
4. Losses printed out are different than with PL, (seem to be normalized in some way). This merits investigation.
cc @patil-suraj

### PR DESCRIPTION
**Pull request by patil-suraj**
# What does this PR do?

This PR adds the ability to configure `lr_scheduler` from command line for `Seq2SeqTrainer`.

Fixes #7543

@sshleifer

### CODE REVIEW

File: examples/seq2seq/seq2seq_trainer.py [hunk 1]
```diff
@@ -62,9 +81,19 @@ def create_optimizer_and_scheduler(self, num_training_steps: int):
                 )
 
         if self.lr_scheduler is None:
-            self.lr_scheduler = get_linear_schedule_with_warmup(
+            self.lr_scheduler = self._get_lr_scheduler(num_training_steps)
+
+    def _get_lr_scheduler(self, num_training_steps):
+        get_schedule_func = arg_to_scheduler[self.args.lr_scheduler]
+        if self.args.lr_scheduler not in ["constant", "constant_w_warmup"]:
```
Comment (sshleifer) [inline examples/seq2seq/seq2seq_trainer.py:88]: (nit) slightly cleaner to check positive conditions

if constant: ...
elif constant_w_warmup: ...
else: ...

File: examples/seq2seq/seq2seq_trainer.py [hunk 2]
```diff
@@ -62,9 +81,19 @@ def create_optimizer_and_scheduler(self, num_training_steps: int):
                 )
 
         if self.lr_scheduler is None:
```
Comment (sshleifer) [inline examples/seq2seq/seq2seq_trainer.py:83]: I know it could be set already, but I would rather just warn if it is not None and then overwrite no matter what?
Or am I missing something?

File: examples/seq2seq/seq2seq_trainer.py [hunk 3]
```diff
@@ -62,9 +81,19 @@ def create_optimizer_and_scheduler(self, num_training_steps: int):
...
-            self.lr_scheduler = get_linear_schedule_with_warmup(
+            self.lr_scheduler = self._get_lr_scheduler(num_training_steps)
+
+    def _get_lr_scheduler(self, num_training_steps):
+        get_schedule_func = arg_to_scheduler[self.args.lr_scheduler]
```
Comment (sshleifer) [inline examples/seq2seq/seq2seq_trainer.py:87]: get_schedule_func -> schedule_func

File: examples/seq2seq/seq2seq_trainer.py (hunk 2 above)
Comment (patil-suraj) [reply examples/seq2seq/seq2seq_trainer.py:83]: why should  we overwrite ? `Trainer` allows you pass to `optimizer` and `scheduler` if you need something different from default, so I if passed I can think we shouldn't overwrite

Comment (sshleifer) [reply examples/seq2seq/seq2seq_trainer.py:83]: ok well then we are overwriting/ignoring the `--lr_scheduler` arg. Fine with me. But probably worth a comment/or warning.

**Review by sshleifer — APPROVED** [review]
Happy to merge with or without comment.
